src/test/scala/com/microsoft/azure/sqldb/spark/utils/SQLServerTestUtils.scala

```diff
@@ -0,0 +1,185 @@
+    return conn
+  }
+
+  private def preFlightChecks(config: Map[String, String]): Boolean = {
+    var writeConfig = Config(config)
+    val conn = getConnection(writeConfig, false)
+    val dbName = writeConfig.get[String](SqlDBConfig.DatabaseName).get
+    if(conn.equals(null)){
+      return false
+    }
+
+    //Check if Server and Database Exist
+    var sql = "SELECT name from master.dbo.sysdatabases where name='"+ dbName + "' OR [name]='"+ dbName + "'"
+    try{
+      val stmt = conn.createStatement()
+      val req = stmt.executeQuery(sql)
+      conn.close()
+      var res = req.getString(1)
+      if(res.eq(null) || res.eq("") || !res.eq(dbName)){
+        return false
+      }
+      return true
+    } catch {
+      case e: Exception => return false
+    }
+  }
+
+  def createTable(config: Map[String, String], columns: Map[String, String]): Boolean ={
+    var writeConfig = Config(config)
+    val table = writeConfig.get[String](SqlDBConfig.DBTable).getOrElse(null)
+    if(table.equals(null)) {
+      return false
+    }
+    val conn = getConnection(writeConfig, true)
+    if(conn.equals(null)){
+      return false
+    }
+    if(!dataTypeVerify(columns)){
+      return false
+    }
+
+    var tableDroppedIfExists = dropTable(config)
+    if(tableDroppedIfExists){
+      var columnDef = ""
+      for(column <- columns){
+        columnDef += column._1 + " " + column._2 + ", "
+      }
+      var sql = "CREATE TABLE " + table + "(" + columnDef.substring(0, columnDef.length-1) + ");"
+      try{
+        val stmt = conn.createStatement()
+        stmt.executeUpdate(sql)
+        conn.close()
+      } catch {
+        case e: Exception => return false
+      }
+    } else {
+      return false
+    }
+
+    return true
+  }
+
+  def dropTable(config: Map[String, String]): Boolean ={
+    var writeConfig = Config(config)
+    val table = writeConfig.get[String](SqlDBConfig.DBTable).getOrElse(null)
+    if(table.equals(null)) {
+      return false
+    }
+    val conn = getConnection(writeConfig, true)
+    if(conn.equals(null)){
+      return false
+    }
+
+    // var sql = "IF EXISTS(SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = '" + table + "') DROP TABLE " + table + ";"
+    var sql = "DROP TABLE IF EXISTS " + table   //Supported SQL Server 2016 and beyond
+    try{
+      val stmt = conn.createStatement()
+      stmt.executeUpdate(sql)
+      conn.close()
+    } catch {
+      case e: Exception => return false
+    }
+
+    return true
+
+  }
+
+  def truncateTable(config: Map[String, String]): Boolean ={
+    var writeConfig = Config(config)
+    val table = writeConfig.get[String](SqlDBConfig.DBTable).getOrElse(null)
+    if(table.equals(null)) {
+      return false
+    }
+    val conn = getConnection(writeConfig, true)
+    if(conn.equals(null)){
+      return false
+    }
+
+    var sql = "TRUNCATE TABLE " + table + ";"
```

Comment:
This is perhaps the first case where we are executing dynamically constructed SQL statement. A general good practice here would be to handle spaces and special characters within the 'table' variable by surrounding that within []. In this case it is probably okay given this is test code but in general you should watch out for strange characters in SQL object identifiers when constructing SQL like this.